curves/ed_on_bls12_381_bandersnatch/src/curves/mod.rs

```diff
@@ -163,6 +162,77 @@ impl Elligator2Config for BandersnatchConfig {
 }
 
+// Constants used in mapping TE form to SW form and vice versa
+//
+// sage: q = 52435875175126190479447740508185965837690552500527637822603658699938581184513
+// sage: Fq = GF(q)
+// sage: MONT_A = 29978822694968839326280996386011761570173833766074948509196803838190355340952
+// sage: MONT_B = 25465760566081946422412445027709227188579564747101592991722834452325077642517
+// sage: MONT_A/Fq(3)
+// 9992940898322946442093665462003920523391277922024982836398934612730118446984
+// sage: Fq(1)/MONT_B
+// 41180284393978236561320365279764246793818536543197771097409483252169927600582
+const MONT_A_OVER_THREE: Fq =
+    MontFp!("9992940898322946442093665462003920523391277922024982836398934612730118446984");
+const MONT_B_INV: Fq =
+    MontFp!("41180284393978236561320365279764246793818536543197771097409483252169927600582");
+
+impl BandersnatchConfig {
+    /// Map an point in TE form to its corresponding  point on SW curve
+    pub fn map_te_to_sw(point: EdwardsAffine) -> Option<SWAffine> {
+        //First we map the point from the TE to Montgamory
+        //edward to mont  ((1+y)/(1-y), (1+y)/(x(1-y)))
+        let v_denom = <<BandersnatchConfig as CurveConfig>::BaseField as One>::one() - point.y;
+        let w_denom = point.x - point.x * point.y;
+
+        let v_denom_inv = v_denom.inverse()?;
+        let w_denom_inv = w_denom.inverse()?;
+
+        let v_w_num = <<BandersnatchConfig as CurveConfig>::BaseField as One>::one() + point.y;
+
+        let v = v_w_num * v_denom_inv;
+        let w = v_w_num * w_denom_inv;
+
+        //now  we are mappyng the montgamory point to SW.
+        //Mont->SW ((x+A/3)/B,y/B)
+
+        let x = MONT_B_INV * (v + MONT_A_OVER_THREE);
+        let y = MONT_B_INV * w;
+
+        let point_on_sw_curve = SWAffine::new_unchecked(x, y);
+        debug_assert!(
+            point_on_sw_curve.is_on_curve(),
+            "TE point mapped off the SW curve"
+        );
+
+        return Some(point_on_sw_curve);
+    }
+
+    pub fn map_sw_to_te(point: SWAffine) -> Option<EdwardsAffine> {
+        //first  we are mappyng the sw point to  montgamory point
+        //SW->Mont by (Bx−A/3,By)
+        let mx = <BandersnatchConfig as MontCurveConfig>::COEFF_B * point.x - MONT_A_OVER_THREE;
+        let my = <BandersnatchConfig as MontCurveConfig>::COEFF_B * point.y;
+
+        //Then we map the TE point to Montgamory
+        // (x,y)↦(x/y,(x−1)/(x+1))
+        let v_denom = my.inverse()?;
+        let x_p_1 = mx + <<BandersnatchConfig as CurveConfig>::BaseField as One>::one();
+        let w_denom = x_p_1.inverse()?;
+
+        let v = mx * v_denom;
+        let w = (mx - <<BandersnatchConfig as CurveConfig>::BaseField as One>::one()) * w_denom;
+
+        let point_on_te_curve = EdwardsAffine::new_unchecked(v, w);
+        debug_assert!(
+            point_on_te_curve.is_on_curve(),
+            "TE point mapped off the SW curve"
+        );
```

Comment:
Same comment as above: consider using `EdwartdsAffine::new` instead and ditch the debug assert.